.github/workflows/Image-build-AB-diary.yaml

```
@@ -1,119 +1,194 @@
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: mvn -B -f backend/ab-diary/pom.xml package org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.1.2184:sonar -Dsonar.projectKey=solutions_ab_diary -Dsonar.organization=solutions -Dsonar.host.url=https://sonarcloud.io/
-
-      # Login to Docker Registry
-      - name: Login to Docker Registry
-        run: docker login -u venkatramireddyb -p ${{ secrets.ACR_PASSWORD }}
-
-      - name: Set Version from Tag or Branch
-        id: set_version
+      - name: Set github branch
+        id: set_branch
         run: |
```

actionlint (1.7.9)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)